### PR TITLE
✨ adds title to portrait component in amp-story-bookend and change expected fields in articles

### PR DESCRIPTION
**Pull request by Enriqe**
* add title to portrait component
* make `title` **required** in `portrait` and `landscape` components
* make `category` **optional** in `portrait` and `landscape` components

Should be added in documentation (#15465) too.

### CODE REVIEW

File: extensions/amp-story/1.0/test/test-amp-story-bookend.js [hunk 1]
```diff
@@ -130,6 +139,9 @@ describes.realWin('amp-story-bookend', {
     registerServiceBuilder(win, 'localization', () => localizationService);
 
     bookend = new AmpStoryBookend(bookendElem);
+
+    // Force sync mutateElement.
+    sandbox.stub(bookend, 'mutateElement').callsArg(0);
```
Comment (Enriqe) [inline extensions/amp-story/1.0/test/test-amp-story-bookend.js]: @newmuis this fixed the `this.element.getResources is not a funciton` error for me

File: extensions/amp-story/1.0/bookend/components/portrait.js [hunk 2]
```diff
@@ -50,9 +52,10 @@ export class PortraitComponent {
 
   /** @override */
   assertValidity(portraitJson, element) {
-    user().assert('category' in portraitJson && 'image' in portraitJson &&
-      'url' in portraitJson, 'Portrait component must contain `category`, ' +
-      '`image`, and `url` fields, skipping invalid.');
+    user().assert('category' in portraitJson && 'title' in portraitJson &&
```
Comment (alanorozco) [inline extensions/amp-story/1.0/bookend/components/portrait.js]: Consider:
```js
const requiredFields = ['category', 'title', 'image', 'url'];
user().assert(
     !requiredFields.some(field => !(field in portraitJson)),
    `Portrait component must contain ' + requiredFields.map(f => '`' + f + '`').join(', ') +
      'fields, skipping invalid.');
```

File: extensions/amp-story/1.0/test/test-amp-story-bookend.js (hunk 1 above)
Comment (alanorozco) [reply extensions/amp-story/1.0/test/test-amp-story-bookend.js]: TIL about `callsArg`!

**Comment by Enriqe** [comment]
@newmuis @alanorozco @gmajoulet PTAL

**Comment by Enriqe** [comment]
ready to merge @newmuis

**Pull request merged**